lib/seahorn/HornClauseDB.cc

```diff
@@ -31,6 +36,67 @@ namespace seahorn
 
     }
   }
+
+  void HornClauseDB::buildCallGraph () 
+  {
+    // indexes must be first computed
+    buildIndexes ();
+    
+    for (auto p: getRelations ()) 
+    {
+      // -- callees
+      expr_set_type callees;
+      const horn_set_type& uses = use(p);
+      for (const HornRule* r: uses)
+      { callees.insert(bind::fname(r->head())); }
+      m_callees.insert(std::make_pair(p, callees));
+
+      // -- callers
+      expr_set_type callers;
+      const horn_set_type& defs = def(p);
+      for (const HornRule* r: defs)
+        filter (r->body (), HornClauseDB::IsRelation(*this), 
+                std::inserter(callers, callers.begin())); 
+      m_callers.insert(std::make_pair(p, callers));
+
+      LOG("horn-cg", 
+          errs () << *(bind::fname(p)) << "\n"
+                  << "\tNumber of callers=" << callers.size() << "\n"
+                  << "\tNumber of callees=" << callees.size() << "\n";
+          if (!callers.empty()) {
+            errs () << "\tCALLERS=";
+            for (auto c: callers) {
+              errs () << *(bind::fname(c)) << "  ";
+            }
+            errs () << "\n";
+          }
+          if (!callees.empty()) {
+            errs () << "\tCALLEES=";
+            for (auto c: callees) {
+              errs ()  << *(bind::fname(c)) << "  ";
+            }
+            errs () << "\n";
+          });
+    }
+
+    // XXX: we are looking for a predicate that corresponds to the
+    // entry block of main. It is not enough to search for a predicate
+    // with no callers since predicates from each function entry won't
+    // have callers.
+    for (auto p: getRelations ()) 
+    {
+      Expr fname = bind::fname(p);
+      std::string sname = boost::lexical_cast<std::string> (fname);
+      if (boost::starts_with(sname, "main") && callers(p).size() == 0) { 
```

Comment:
There should be a better way. `main` is not always the entry point. 
This is ok as a workaround, as long as it is properly exposed. That is, not part of a generic call-graph functionality.